clash-prelude/src/Clash/Signal.hs

```diff
@@ -1045,16 +1049,58 @@ sampleN
    . ( KnownDomain dom
      , Undefined a )
   => Int
-  -- ^ The number of samples we want to see
-  -> (HiddenClockResetEnable dom  => Signal dom a)
-  -- ^ 'Signal' we want to sample, whose source potentially has a hidden clock
+  -- ^ Number of samples to produce
+  -> (HiddenClockResetEnable dom => Signal dom a)
+  -- ^ 'Signal' to sample, whose source potentially has a hidden clock
   -- (and reset)
   -> [a]
 sampleN n s0 =
   let s1 = exposeClockResetEnable s0 clockGen (resetGen @dom) enableGen in
   S.sampleN n s1
 {-# NOINLINE sampleN #-}
 
+-- | Get a list of samples from a 'Signal', while asserting the reset line
+-- for /n/ clock cycles. 'sampleReset' does not return the first /n/ cycles,
+-- i.e., when the reset is asserted.
+--
+-- __NB__: This function is not synthesizable
+sampleReset
+  :: forall dom a n
+   . ( KnownDomain dom
+     , Undefined a
+     , 1 <= n )
+  => SNat n
+  -- ^ Number of cycles to assert the reset
+  -> (HiddenClockResetEnable dom => Signal dom a)
+  -- ^ 'Signal' to sample, whose source potentially has a hidden clock
+  -- (and reset)
+  -> [a]
+sampleReset nReset f0 =
+  let f1 = exposeClockResetEnable f0 clockGen (resetGenN @dom nReset) enableGen in
+  drop (snatToNum nReset) (S.sample f1)
+{-# NOINLINE sampleReset #-}
+
+-- | Get a fine list of /m/ samples from a 'Signal', while asserting the reset line
+-- for /n/ clock cycles. 'sampleReset' does not return the first /n/ cycles,
+-- i.e., while the reset is asserted.
+--
+-- __NB__: This function is not synthesizable
+sampleResetN
+  :: forall dom a n
+   . ( KnownDomain dom
+     , Undefined a
+     , 1 <= n )
+  => SNat n
+  -- ^ Number of cycles to assert the reset
+  -> Int
+  -- ^ Number of samples to produce
+  -> (HiddenClockResetEnable dom => Signal dom a)
+  -- ^ 'Signal' to sample, whose source potentially has a hidden clock
+  -- (and reset)
+  -> [a]
+sampleResetN nReset nSamples f =
```

Comment:
I think the `N`, `n` and `m` are somewhat confusing.
Because the `N` in the name is connected to the `m`, and not the `n`.

Reply:
Maybe changing the `n` to `r` is clearer?

Reply:
Yeah, good idea.

Reply:
To keep it consistent, you might want to do the same to `sampleReset`